llama.cpp

```diff
@@ -6781,6 +6818,153 @@ struct llm_build_context {
         return gf;
     }
 
+    // ref: https://arxiv.org/abs/2203.03466
+    //      https://github.com/ggerganov/llama.cpp/issues/5276#issuecomment-1925774738
+    // based on the original build_llama() function
+    struct ggml_cgraph * build_minicpm() {
+        struct ggml_cgraph * gf = ggml_new_graph_custom(ctx0, LLAMA_MAX_NODES, false);
+
+        const int64_t n_embd_head = hparams.n_embd_head_v;
+        GGML_ASSERT(n_embd_head == hparams.n_embd_head_k);
+        GGML_ASSERT(n_embd_head == hparams.n_rot);
+
+        const int64_t n_embd = hparams.n_embd;
+        //TODO: if the model varies, these parameters need to be read from the model
+        const int scale_emb = 12;
+        const int dim_model_base = 256;
+        const float scale_depth = 1.4f;
```

Comment:
Minor change of the constant names:

```suggestion
        const int64_t n_embd_base = 256;
        const float scale_embd  = 12.0f;
        const float scale_depth = 1.4f;
```

Reply:
> Minor change of the constant names:

Done with corresponding modifications: [this commit](https://github.com/ggerganov/llama.cpp/pull/5346/commits/390996d84c3af4d71dc272c18b5793123a956e92)